hal/spi/spi_drv_stm32.c

```diff
@@ -90,6 +90,20 @@ static void spi1_pins_setup(void)
     SPI_PIO_AFL = reg | (SPI1_PIN_AF << ((SPI1_MOSI_PIN) * 4));
     reg = SPI_PIO_AFL & ~(0xf << ((SPI1_MISO_PIN) * 4));
     SPI_PIO_AFL = reg | (SPI1_PIN_AF << ((SPI1_MISO_PIN) * 4));
+
```

Comment:
These changes don't work on STM32F4. I suggest to move these settings to the STM32L0 hal if they are needed for SPI to work on that platform.